### DISCUSSION TITLE
How to access setValue from outside the form?

### DISCUSSION THREAD

**Discussion by rizabek-mukanov** [react-hook-form/react-hook-form#4225] [Q&A]
I have a form builder component for whole project based on react-hook-form. So, everywhere i access it from outside.
Now i have a need to programmatically call setValue from outside of form, hook in diagram below.
links below, hope it helps to understand

[link to screenshot of diagram](https://ibb.co/tHTBkMp)
[link to screenshot of form fields](https://ibb.co/bm7vrgY)
[link to screenshot of rhf devtools](https://ibb.co/992TYyZ)

**Comment by bluebill1049** [comment]
you can have it outside of `useForm`, you may have to move `useForm` upper at your component level.

**Comment by guilhermelim** [comment]
Can you give more details on how to call setValue from outside the form?

**Reply by Moshyfawn** [reply]
You can't really. You need to init the form in order to get its methods. You might wanna rethink / restructure your form instead

**Reply by rizabek-mukanov** [reply]
@guilhermelim if i remember correctly we somehow passed setValue function somewhere else from form elements. I dont have access to codebase since it was on previous job. Otherwise it is also possible we changed structure as @bluebill1049 and @Moshyfawn suggested. 
Can't give you more details as i dont remember.

**Comment by poznianski** [comment]
Right now I moved useForm to my useContext and shared all I need between components, including setValue. Works perfect.
